src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SegmentFilters.cs

```diff
@@ -128,5 +128,50 @@ public static bool HasSegments(Hl7v2Data hl7v2Data, string segmentIdContent)
 
             return result;
         }
+
+        public static List<Hl7v2Data> SliceDataBySegments(Hl7v2Data hl7v2Data, List<Hl7v2Segment> segmentList, Hl7v2Segment endSegment = null)
+        {
+            var results = new List<Hl7v2Data>();
+            var result = new Hl7v2Data();
+            string[] segmentSeparators = { "\r\n", "\r", "\n" };
+            var valueList = hl7v2Data.Value.Split(segmentSeparators, StringSplitOptions.RemoveEmptyEntries);
+            var startSegmentFound = false;
+            var segmentIndex = 0;
+            for (var i = 0; i < hl7v2Data.Meta.Count; ++i)
+            {
+                if (segmentIndex == 0 && segmentList.Count > 0 && ReferenceEquals(hl7v2Data.Data[i], segmentList[segmentIndex]))
```

Comment:
just curious, why do we use ReferenceEquals here?

Reply:
There may be two identical elements in the `segmentList`. The reason for using it is not only to require that the values of the objects are equal, but also to ensure that the references of the two objects are equal.

Reply:
why not to use "=="?